pySDC/projects/DAE/sweepers/RungeKuttaDAE.py

```diff
@@ -0,0 +1,181 @@
+
+    can be implemented as follows:
+
+    >>> class newRungeKuttaMethod(RungeKutta):
+    >>>     nodes = np.array([c1, c2, c3])
+    >>>     weights = np.array([b1, b2, b3])
+    >>>     matrix = np.zeros((3, 3))
+    >>>     matrix[0, 0] = a11
+    >>>     matrix[1, :2] = [a21, a22]
+    >>>     matrix[2, :] = [a31, a32, a33]
+    >>>     ButcherTableauClass = ButcherTableau
+
+    The new class ``newRungeKuttaMethodDAE`` can then be used by defining the DAE class inheriting from both, this base class and class containing
+    the Butcher tableau:
+
+    >>> class newRungeKuttaMethodDAE(RungeKuttaDAE, newRungeKuttaMethod):
+    >>>     pass
+
+    More details can be found [here](https://github.com/Parallel-in-Time/pySDC/blob/master/pySDC/implementations/sweeper_classes/Runge_Kutta.py).
+    """
+
+    def __init__(self, params):
+        super().__init__(params)
+        self.du_init = None
+
+    def predict(self):
+        """
+        Predictor to fill values at nodes before first sweep.
+        """
+
+        # get current level and problem
+        lvl = self.level
+        prob = lvl.prob
+
+        du_init = self.du_init[:] if self.du_init is not None else prob.du_exact(lvl.time)[:]
+        lvl.f[0] = prob.dtype_f(du_init)
+        for m in range(1, self.coll.num_nodes + 1):
+            lvl.u[m] = prob.dtype_u(init=prob.init, val=0.0)
+            lvl.f[m] = prob.dtype_f(init=prob.init, val=0.0)
+
+        lvl.status.unlocked = True
+        lvl.status.updated = True
+
+    def update_nodes(self):
+        r"""
+        Updates the values of solution ``u`` and their gradient stored in ``f``.
+        """
+
+        # get current level and problem description
+        lvl = self.level
+        prob = lvl.prob
+
+        # only if the level has been touched before
+        assert lvl.status.unlocked
+        assert lvl.status.sweep <= 1, "RK schemes are direct solvers. Please perform only 1 iteration!"
+
+        M = self.coll.num_nodes
+        for m in range(M):
+            u_approx = prob.dtype_u(lvl.u[0])
+            for j in range(1, m + 1):
+                u_approx += lvl.dt * self.QI[m + 1, j] * lvl.f[j][:]
+
+            def F(du):
+                r"""
+                This function builds the implicit system to be solved for a DAE of the form
+
+                .. math::
+                    0 = F(u, u', t)
+
+                Applying a RK method yields the (non)-linear system to be solved
+
+                .. math::
+                    0 = F(u_0 + \Delta t \sum_{j=1}^m a_{ij} U_j, U_m, \tau_m),
+
+                which is solved for the derivative of u.
+
+                Parameters
+                ----------
+                du : dtype_u
+                    Unknowns of the system (derivative of solution u).
+
+                Returns
+                -------
+                sys : dtype_f
+                    System to be solved.
+                """
+                du_mesh = prob.dtype_f(du)
```

Comment:
Why is this called `du_mesh` and not `local_du_approx` or something more descriptive?